R/functions__eager.R

```diff
@@ -37,41 +38,89 @@
 #' # diagonal
 #' pl$concat(l_hor, how = "diagonal")
 pl$concat = function(
-    l, # list of DataFrames or Series or lazyFrames or expr
+    ..., # list of DataFrames or Series or lazyFrames or expr
     rechunk = TRUE,
-    how = c("vertical", "horizontal", "diagonal"),
-    parallel = TRUE # not used yet
-    ) {
+    how = c("vertical", "horizontal", "diagonal"), # , "vertical_relaxed","diangonal_relaxed"),
+    parallel = TRUE,
+    # eager = FALSE,
+    to_supertypes = FALSE) {
+  # unpack arg list
+  l = unpack_list(..., skip_classes = "data.frame")
+
+  # nothing becomes NULL
+  if (length(l) == 0L) {
+    return(NULL)
+  }
```

Comment:
I guess we should error instead in this situation, this can only come from a user input misspecification

Reply:
how about when using pl$concat  "dynamically" ?

```R
lf_all_logs_that_day =
  list.files("./error_logs/2022-01-01/", full = TRUE, pattern = "*.parquet") |>
  lapply(pl$scan_parquet) |>
  pl$concat(how = "vertical")
```

it could be there was no error log parquet files on that specific day

```R
> cbind()
NULL
> rbind()
NULL
> 
```

Reply:
I personally prefer when an API take into account the zero case, because it more tedious to code around. But if you prefer Error, I'll remove it again.

Reply:
`data.table::rbindlist()` also returns `NULL` when we provide a list of `NULL`s so it's fine to let this as-is